appserver/web/web-core/src/main/java/org/apache/catalina/connector/CoyoteAdapter.java

```diff
@@ -687,10 +696,10 @@ private static boolean normalizeChars(MessageBytes uriMessageBytes) {
             }
             // Prevent from going outside our context
             if (index == 0) {
-                return false;
+                throw new IOException("Request traversed outside of allowed context");
             }
             int index2 = -1;
-            for (pos = start + index - 1; (pos >= 0) && (index2 < 0); pos--) {
+            for (pos = start + index - 1; (pos >= 0) && (index2 < 0); pos --) {
```

Comment:
I'd rather keep `--` operator not separated, like in new line 604 is.

Reply:
That will have been a cherry-pick reformatting which I missed, will fix.